h5c/vic/src/vic-webapp/src/app/vch-view/vch-view.component.ts

```diff
@@ -225,6 +231,50 @@ export class VicVchViewComponent implements OnInit, OnDestroy {
     this.vmViewService.getContainersData({}); // TODO: filter containers for the vch
   }
 
+   /**
+   * Download Certificate for VCH
+   */
+  downloadVchCert(vch: VirtualContainerHost) {
+    this.isDgLoading = true;
+    Observable.combineLatest(
+      this.createWzService.getVicApplianceIp(),
+      this.createWzService.acquireCloneTicket()
+    ).catch(err => {
+      return Observable.throw(err);
+    }).subscribe(([serviceHost, cloneTicket]) => {
+      const vchId = vch.id.split(':')[3];
+      const servicePort = VIC_APPLIANCE_PORT;
+      const targetHost = this.extSessionService.getVcenterServersInfo()[0];
+      const targetHostname = targetHost.name;
+      const targetThumbprint = targetHost.thumbprint;
```

Comment:
Not a blocker for 1.3.0, but... can we file an issue to clean this up after?

It seems like the code here is duplicated in at least two other places already:
```
     const vchId = vch.id.split(':')[3];
     const servicePort = VIC_APPLIANCE_PORT;
     const targetHost = this.extSessionService.getVcenterServersInfo()[0];
     const targetHostname = targetHost.name;
     const targetThumbprint = targetHost.thumbprint; 
```

Having a central place where we construct URLs for making requests to the API in a consistent way should both simplify things and reduce fragility (which will be important as we haven't committed to maintaining backwards-compatibility with the current API in future releases).

Reply:
Agreed. Will create a refactoring issue for these repetitive functions.